arangod/Aql/ExecutionBlock.cpp

```diff
@@ -136,40 +133,44 @@ void ExecutionBlock::collectExecStats(ExecutionStats& stats) {
 }
 
 void ExecutionBlock::traceExecuteBegin(AqlCallStack const& stack,
-                                       std::string const& clientId) {
+                                       std::string_view clientId) {
+  // add timing for block in case profiling is turned on.
+
+  // intentionally negative so it is easier to spot calculation errors.
+  double start = -1.0;
+  auto profilingGuard = scopeGuard([&]() noexcept {
+    _execNodeStats.runtime += currentSteadyClockValue() - start;
+  });
   if (_profileLevel >= ProfileLevel::Blocks) {
-    if (_execNodeStats.runtime >= 0.0) {
-      _execNodeStats.runtime -= currentSteadyClockValue();
-      TRI_ASSERT(_execNodeStats.runtime < 0.0);
-    }
+    // only if profiling is turned on, get current time
+    start = currentSteadyClockValue();
+  } else {
+    // if profiling is turned off, don't get current time, but instead
+    // cancel the scopeGuard so we don't unnecessarily call timing
+    // functions on exit
+    profilingGuard.cancel();
```

Comment:
That won't work like that - we cannot use a scopeGuard here, but do the update in `traceExecuteEnd`.